core/src/main/java/io/questdb/cairo/pool/WriterPool.java

```diff
@@ -74,6 +72,7 @@ public class WriterPool extends AbstractPool {
     private static final Log LOG = LogFactory.getLog(WriterPool.class);
     private static final long QUEUE_PROCESSING_OWNER = -2L;
     private final MicrosecondClock clock;
+    private final LongObjHashMap<CommitListener> commitListeners = new LongObjHashMap<CommitListener>();
```

Comment:
This is not thread-safe collection, can be problem if used to get and set at the same time.

Reply:
At the moment it's only added-to by AccessListStore. Should I replace it with concurrent hash map ?

Reply:
i think what we need to do is change setCommitListener() to this:
```
    public void setCommitListener(TableToken tt, CommitListener listener) {
        try (TableWriter tw = get(tt, "set commit listener")) {
            assert commitListeners.get(tt.getTableId()) == null;
            commitListeners.put(tt.getTableId(), listener);
            tw.setCommitListener(listener);
        }
    }
```

then set and get are all done under the writer's lock so we should be fine.

Reply:
made the change in latest commit .

Reply:
If different threads will run the method for different tables the hashtable will be destroyed.

As far as I understand these listeners are set at the startup only. I'd recommend renaming the method to 
`setCommitListenerUsafe` and make a comment that this can only be called on a single thread.

Reply:
CAS and fences provide visibility, so threads would see each other's changes.
however, i see your point.
if two threads are adding listeners at the same time, and they hash into the same bucket, there might be a problem.

then we could just change to a concurrent map, as Bolek suggested above.
especially that you are also suggesting to use the table dir as key.
so we could just use `ConcurrentHashMap<CommitListener>` with `tableToken.getDirName()` as key.

Reply:
changed in 0d914af